docs/usage.md

```diff
@@ -0,0 +1,85 @@
+## Usage
+
+### Firmware Flashing
+
+First install the `arduino-fwuploader`. Extract the zip file and, for example, to update the NINA module present on the
+Arduino MKR WiFi 1010, run:
+
+```
+./arduino-fwuploader firmware flash -b arduino:samd:mkrwifi1010 -a /dev/ttyACM0
+```
+
+You just have to specify the fqbn (`-b` or `--fqbn`) of the board and the serial port (`-a` or `--address`) The firmware
+uploader will take care of fetching everything is required to perform the update process. If no module and version are
+specified **the latest version of the firmware** will be used.
+
+If you want to flash a specific version of a firmware you can use the `-m` or `--module` flag
+
+For example to flash the WINC module present on the MKR 1000 with 19.6.1 version of the firmware you can run something
+like:
+
+```
+./arduino-fwuploader firmware flash -b arduino:samd:mkr1000 -a /dev/ttyACM0 -m WINC1500@19.6.1
+```
+
+There is also a retry mechanism bundled in the tool because the flashing process uses serial comunication and sometimes
+can be a bit unreliable. The retry flag is set by default to 9 retries, but it's possible to overwrite it for whatever
+reason. For example to update a Nano RP2040 Connect with the retry set to 2 you can use:
```

Comment:
```suggestion
There is a retry mechanism because the flashing process uses serial communication, which sometimes can be a bit
unreliable. The retry flag is set by default to 9 retries, but it's possible to overwrite it for whatever reason. For
example to update a Nano RP2040 Connect with the retry set to 2 you can use:
```
- Fix "comunication" typo
- Insignificant rewording
- Adjust line wrapping of paragraph after making the above changes.